### PR TITLE
Sc 5770

### PR DESCRIPTION
**Pull request by IonesioJunior**
## Description
Please include a summary of the change, the motivation, and any additional context that will help others understand your PR. If it closes one or more open issues, [please tag them as described here](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

**Review by shubham3121 — COMMENTED** [review]
Awesome stuff !!! @IonesioJunior 🎉 . Left some minor comments.

File: packages/syft/src/syft/core/node/common/client_manager/dataset_api.py [hunk 1]
```diff
@@ -174,6 +174,21 @@ def all_as_datasets(self) -> List[Any]:
...
+        if not skip_check:
+            pref = input(
+                "You are about to delete all datasets ? 🚨 \n"
+                "All information will be permanantely deleted.\n"
+                "Please enter y/n to proceed: "
+            )
+            while pref != "y" and pref != "n":
+                pref = input("Invalid input '" + pref + "', please specify 'y' or 'n'.")
+            if pref == "n":
+                raise Exception("Datasets deletion is cancelled.")
```
[the remark this answers is not in the document]
Comment (IonesioJunior) [reply packages/syft/src/syft/core/node/common/client_manager/dataset_api.py]: Done! [8f0deb7](https://github.com/OpenMined/PySyft/pull/6571/commits/8f0deb7ebaad7a99497cde3046b7a8904951d7b9)

File: tests/integration/redis/dataset_test.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            raise Exception
+
+    domain_owner.datasets[0].delete("test_data", skip_check=True)
+
+    # Check if the number of assets has been decreased
+    assert len(domain_owner.datasets[0].data) == 1
+
+    # Check if the asset key (train_data) is still there
+    for asset in domain_owner.datasets[0].data:
+        if asset["name"] == "test_data":
```
[the remark this answers is not in the document]
Comment (IonesioJunior) [reply tests/integration/redis/dataset_test.py:65]: Done! [9508598](https://github.com/OpenMined/PySyft/pull/6571/commits/9508598adfa3c3ea985146af579355d59f3ce6dd)

File: tests/integration/redis/dataset_test.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+            raise Exception
+
+    domain_owner.datasets[0].delete("val_data", skip_check=True)
+
+    # Check if the number of assets has been decreased
+    assert len(domain_owner.datasets[0].data) == 0
+
+    # Check if the asset key (train_data) is still there
+    for asset in domain_owner.datasets[0].data:
+        if asset["name"] == "val_data":
```
[the remark this answers is not in the document]
Comment (IonesioJunior) [reply tests/integration/redis/dataset_test.py]: Done! [9508598](https://github.com/OpenMined/PySyft/pull/6571/commits/9508598adfa3c3ea985146af579355d59f3ce6dd)

**Review by shubham3121 — APPROVED** [review]
🚀🎉

File: packages/syft/src/syft/core/node/common/client_manager/dataset_api.py [hunk 4]
```diff
@@ -174,6 +174,22 @@ def all_as_datasets(self) -> List[Any]:
...
+                "Please enter y/n to proceed: "
+            )
+            while pref != "y" and pref != "n":
+                pref = input("Invalid input '" + pref + "', please specify 'y' or 'n'.")
+            if pref == "n":
+                print("Datasets deletion is cancelled.")
+                return None
+
+        for dataset in self.all():
+            self.delete(dataset_id=dataset.get("id"))
```
Comment (rasswanth-s) [inline packages/syft/src/syft/core/node/common/client_manager/dataset_api.py:191]: MAYBE: Shall we pass the skip check to .delete function from purge, so if a person passes a skip check, we could automatically skip check for the assets also ?

File: packages/syft/src/syft/core/node/common/client_manager/dataset_api.py [hunk 5]
```diff
@@ -459,15 +475,16 @@ def delete(self, name: str) -> bool:
         if asset_id is None:
             raise KeyError(f"The asset with name `{name}` does not exists.")
```
Comment (rasswanth-s) [inline packages/syft/src/syft/core/node/common/client_manager/dataset_api.py:477]: Q:  L470 , does we delete all the assets in a datasets, I think we are fetching only the first asset_id , as dataset might have many assets?

**Review by rasswanth-s — APPROVED** [review]
Wonderful, Ionesio Heaps of Data Deletion in Zing....

**Pull request merged**